File: FtcMobileAdsExample_OC/FtcMobileAdsExample/ViewController.m
//
//  ViewController.m
//  FtcMobileAdsExample
//
//  Created by KYPoseidonL on 2020/10/16.
//

#import "ViewController.h"
#import <FtcMobileAds/FtcMobileAds.h>

@interface ViewController ()<UITextFieldDelegate, FtcInterstitialDelegate>

@property (weak, nonatomic) IBOutlet UITextField *txtAccountID;
@property (weak, nonatomic) IBOutlet UITextField *txtZoneID;
@property (weak, nonatomic) IBOutlet UITextView *txtLog;
@property (weak, nonatomic) IBOutlet UIButton *btnDismiss;
@property (weak, nonatomic) IBOutlet UIButton *btnDisplayInterstitial;
@property (weak, nonatomic) IBOutlet UIView *bannerContainer;

@property(nonatomic, strong) FtcInterstitial *interstitial;
@property(nonatomic, strong) FtcBanner *ftcBanner;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    self.txtAccountID.delegate = self;
    self.txtZoneID.delegate = self;
}

- (IBAction)onGetBannerClick:(id)sender {
    self.txtAccountID.text = @"180130";
    self.txtZoneID.text = @"457319";
    if(![self validateInputs]){
        return;
    }

    self.btnDismiss.hidden = NO;
    PlacementRequestConfig *config = [[PlacementRequestConfig alloc] initWithAccountId:[self.txtAccountID.text integerValue] zoneId:[self.txtZoneID.text integerValue] width:0 height:0 keywords:nil click:nil];
    [FtcMobileAds requestPlacementWithConfig:config success:^(FtcResponseStatus status, NSArray<Placement *> *placements) {
        NSLog(@"requestPlacementWithConfig: status = %d, placements = %@", status, placements);
        if (status == FtcResponseStatusSuccess) {
            self.ftcBanner = [[FtcBanner alloc] initWithPlacement:placements[0] container:self.bannerContainer placementRequestConfig:config];
        }
    } failure:^(NSError *error) {
        NSLog(@"error = %@", error);
    }];
}

- (IBAction)onDismissClick:(id)sender {
    [self.ftcBanner destroy];
    self.btnDismiss.hidden = YES;
}

- (IBAction)onGetInterstitialClick:(id)sender {
    self.txtAccountID.text = @"180130";
//    self.txtZoneID.text = @"457480";
//    self.txtZoneID.text = @"457320";
    self.txtZoneID.text = @"457543";
    if(![self validateInputs]){
        return;
    }
    self.btnDisplayInterstitial.hidden = YES;
    
    PlacementRequestConfig *config = [[PlacementRequestConfig alloc] initWithAccountId:[self.txtAccountID.text integerValue] zoneId:[self.txtZoneID.text integerValue] width:0 height:0 keywords:nil click:nil];
    [FtcMobileAds requestPlacementWithConfig:config success:^(FtcResponseStatus status, NSArray<Placement *> *placements) {
        NSLog(@"requestPlacementWithConfig: status = %d, placements = %@", status, placements);
        if (status == FtcResponseStatusSuccess) {
            self.interstitial = [[FtcInterstitial alloc] initWithPlacement:placements[0] delegate:self];
        }
    } failure:^(NSError *error) {
        NSLog(@"error = %@", error);
    }];
}

- (IBAction)onDisplayInterstitialClick:(id)sender {
    self.btnDisplayInterstitial.hidden = YES;
    [self.interstitial presentFromRootViewController:self];
}


- (BOOL)textFieldShouldReturn:(UITextField *)textField {
    [textField resignFirstResponder];
    return YES;
}

- (BOOL)validateInputs {
    
    if(self.txtAccountID.text.length < 1){
        [self log:@"Invalid account ID"];
        return NO;
    }
    
    if(self.txtZoneID.text.length < 1){
        [self log:@"Invalid zone ID"];
        return NO;
    }
    return YES;
}

- (void)log:(NSString *)str {
    self.txtLog.text = [NSString stringWithFormat:@"%@\n%@", self.txtLog.text, str];
    NSLog(@"%@", str);
}

#pragma mark - FtcInterstitialDelegate
- (void)interstitialStartLoad:(FtcInterstitial *)interstitial {
    [self log:@"Interstitial :: start load"];
}

- (void)interstitialFailedToLoad:(FtcInterstitial *)interstitial {
    [self log:@"Interstitial :: failed"];
}

- (void)interstitialReady:(FtcInterstitial *)interstitial {
    [self log:@"Interstitial :: ready"];
    self.btnDisplayInterstitial.hidden = NO;
}

- (void)interstitialClosed:(FtcInterstitial *)interstitial {
    [self log:@"Interstitial :: close"];
}
@end

File: FtcMobileAdsExample_Swift/FtcMobileAdsExample/ViewController.swift
//
//  ViewController.swift
//  FtcMobileAdsExample
//
//  Created by KYPoseidonL on 2020/10/16.
//

import UIKit


class ViewController: UIViewController , UITextFieldDelegate, FtcInterstitialDelegate {
    
    @IBOutlet weak var btnGetBanner: UIButton!
    @IBOutlet weak var btnGetInterstitial: UIButton!
    
    @IBOutlet weak var txtAccountID: UITextField!
    @IBOutlet weak var txtZoneID: UITextField!
    
    @IBOutlet weak var txtLog: UITextView!

    @IBOutlet weak var btnDisplayInterstitial: UIButton!
    @IBOutlet weak var btnDismiss: UIButton!
    
    @IBOutlet weak var bannerContainer: UIView!
    
    var accountID:Int!
    var zoneID:Int!

    var interstitialReady:Bool = false
    
    var interstitial :FtcInterstitial!
    var ftcBanner :FtcBanner!

    override func viewDidLoad() {
        super.viewDidLoad()
        txtAccountID.delegate = self
        txtZoneID.delegate = self
    }

    @IBAction func onGetBannerClick(_ sender: Any) {
        txtAccountID.text = "180130"
        txtZoneID.text = "457319"
        if(!validateInputs(includePublisher:false)){
            return
        }
        btnDismiss.isHidden = false
        let config: PlacementRequestConfig = PlacementRequestConfig.init(accountId: self.accountID, zoneId: self.zoneID, width: 0, height: 0, keywords: [], click: "")
         FtcMobileAds.requestPlacement(with: config, success: { (status, placements) in
             print(placements!)
             if (status == FtcResponseStatus.success) {
                self.ftcBanner = FtcBanner.init(placement: placements?[0] as! Placement, container: self.bannerContainer, placementRequestConfig: config)
             }
         }) { (error) in
             print(error!)
         }
    }
    
    @IBAction func onGetInterstitialClick(_ sender: Any) {
        txtAccountID.text = "180130"
//        txtZoneID.text = "457480"
//        txtZoneID.text = "457320"
        txtZoneID.text = "457543"
        if(!validateInputs(includePublisher:false)){
            return
        }
        
        let config: PlacementRequestConfig = PlacementRequestConfig.init(accountId: self.accountID, zoneId: self.zoneID, width: 0, height: 0, keywords: [], click: "")
        FtcMobileAds.requestPlacement(with: config, success: { (status, placements) in
            print(placements!)
            if (status == FtcResponseStatus.success) {
                self.interstitial = FtcInterstitial.init(placement: placements?[0] as! Placement, delegate: self)
            }
        }) { (error) in
            print(error!)
        }
    }
    
    @IBAction func onDisplayInterstitialClick(_ sender: Any) {
        btnDisplayInterstitial.isHidden = true
        self.interstitial.present(fromRootViewController: self)
    }
    @IBAction func onDismissClick(_ sender: Any) {
        self.ftcBanner.destroy()
        btnDismiss.isHidden = true
    }
    
    func validateInputs(includePublisher:Bool) -> Bool{
         let accountText = txtAccountID.text
         if(accountText == nil || accountText!.isEmpty){
             log("Invalid account ID")
             return false
         }else{
             let accountID = Int(accountText!) ?? 0
             if(accountID == 0){
                 log("Invalid account ID")
                 return false
             }else{
                 self.accountID = accountID
             }
         }
         
         let zoneText = txtZoneID.text
         if(zoneText == nil || zoneText!.isEmpty){
             log("Invalid zone ID")
             return false
         }else{
             let zoneID = Int(zoneText!) ?? 0
             if(zoneID == 0){
                 log("Invalid zone ID")
                 return false
             }else{
                 self.zoneID = zoneID
             }
         }
     
         return true
     }
    
    // DELEGATE FUNCTIONS
    func log(_ str:String){
        txtLog.text = "> " + str + "\n" + txtLog.text
        NSLog(str) // also log to xcode
    }
    
    func interstitialReady(_ interstitial: FtcInterstitial) {
        log("Interstitial :: ready")
        btnDisplayInterstitial.isHidden = false
    }
    
    func interstitialFailed(toLoad interstitial: FtcInterstitial) {
        log("Interstitial :: failed")
    }
   
    func interstitialClosed(_ interstitial: FtcInterstitial) {
        log("Interstitial :: close")
    }
    func interstitialStartLoad(_ interstitial: FtcInterstitial) {
        log("Interstitial :: start load")
    }
    
    func textFieldShouldReturn(_ textField: UITextField) -> Bool{
        textField.resignFirstResponder()
        return true
    }
}



